doom-modeline-segments.el

```diff
@@ -958,8 +994,12 @@ mouse-3: Next error"
                                      ((> severity note-level)    (cl-incf .warning))
                                      (t                          (cl-incf .note))))))
                         (if (> (+ .error .warning .note) 0)
-                            (doom-modeline-check-icon "nf-md-alert_circle_outline" "❗" "!"
-                                                        (cond ((> .error 0) 'doom-modeline-urgent)
+                            (doom-modeline-check-icon (nth 0 doom-modeline-check-error-icons)
```

Comment:
[L1169](https://github.com/seagle0128/doom-modeline/pull/711/files#diff-0505629a410f96e739a924cc71c2ef537a25d27a64d97615e8e7ae3d06906130R1169) `doom-modeline-display-text` may cause the incorrect displays of the icons. Did you test it?

Reply:
Yes, i tested it, it works well for me, especially with flymake.
![Sin título](https://github.com/seagle0128/doom-modeline/assets/163072436/768ada93-28ad-4ddb-944d-30334f45a15c)

Reply:
The screenshot looks in TUI. How about GUI?

Reply:
Of course `doom-modeline-check-icon` is a good option if it's added. But we should consider the relationship of other options.

Reply:
then Should i use `doom-modeline-icon` instead `doom-modeline-check-icon`?

If so, i should add an option for choose nerd icon kind.

Reply:
We can decide this later.